src/goto-symex/goto_symex_state.cpp

```diff
@@ -788,12 +788,62 @@ void goto_symex_statet::rename_address(
   }
 }
 
+static bool needs_renaming(
+  const typet &type,
+  const namespacet &ns)
+{
+  if(type.id()==ID_array)
+  {
+    const auto &array_type=to_array_type(type);
+    return needs_renaming(array_type.subtype(), ns) ||
+           !array_type.size().is_constant();
```

Comment:
This case, and also the struct/union case are only almost the same as the actual renaming function. Such subtle differences look dangerous.

Reply:
Fixed by means of an extra commit that first does cleanup.